django/db/models/options.py

```diff
@@ -193,6 +193,18 @@ def contribute_to_class(self, cls, name):
             self.unique_together = normalize_together(self.unique_together)
             self.index_together = normalize_together(self.index_together)
 
+            # check for the %(class)s escape used for inherited permissions.
+            # If present, replace it with the appropriate text based off of
+            # the class name for both the name and codename of the permission.
+            perms = meta_attrs.pop('permissions', self.permissions)
+            translated_perms = []
+            if perms:
+                for codename, name in perms:
+                    codename = codename % {'class': cls.__name__.lower()}
```

Comment:
`cls.__name__.lower()` is the same as `self.model_name`. I guess `model_name` would probably be a better placeholder name.